tensorflow_estimator/python/estimator/estimator.py

```diff
@@ -1036,15 +1036,53 @@ def _validate_features_in_predict_input(self, result):
                       'QueueRunner. That means predict yields forever. '
                       'This is probably a mistake.')
 
-  def _get_iterator_from_input_fn(self, input_fn, mode, distribution=None):
-    if distribution is not None:
+  def _has_input_context_arg(self, input_fn_args):
+    # Only support 4 arguments at most. They are `mode`, `config`, `params`
+    # and `input_context` for input_fn. The additional arg will be taken
+    # as the `input_context`. It should be the first one and it is useful 
+    # for distributio strategy.
+    optional_args = tuple(['mode', 'params', 'config'])
```

Comment:
All of this input arg parsing should be in one place; shouldn't _wrap_optional_args... handle this?